subworkflows/local/benchmarking.nf

```diff
@@ -0,0 +1,122 @@
+    OVERLAP_JSON(OVERLAP_PLOT.out.plots, "Overlap plots", "Plot the overlap circRNAs found in total and polyA data for the tools")
+
+    //Location plot workflow
+    LOCATION_PLOT(ch_joined)
+    LOCATION_JSON(LOCATION_PLOT.out.plots, "Location plots", "Plots the location of the circRNAs found" )
+
+
+    //Pearson correllation workflow
+    ch_meta = ch_real_bam.map { it[0] }
+    ch_path = ch_real_bam.map { it[1] }
+    ch_scale = Channel.value(1)
+    ch_genomecov_inputs = ch_meta.combine(ch_path).combine(ch_scale)
+        .map { meta, path, scale ->
+            tuple(meta, path, scale)
+        }
+
+    ch_genomecov = BEDTOOLS_GENOMECOV(ch_genomecov_inputs, [], "bg",false)
+
+    ch_seqdepths = ch_genomecov.genomecov
+        .map { genomecov_result -> genomecov_result[1].toString() }
+        .collectFile(name: 'genomecov_paths.txt',
+                        newLine: true)
+
+    ch_corr = SEQ_DEPTH_CORRELLATION(ch_real_bed, ch_seqdepths.collect()).report
+
+    ch_pearson = ch_corr.splitCsv(header: true, sep: "\t")
+    .map{ values -> [values.tool, values.pearson_corr]}
+    .collectFile( newLine: true,
+                    storeDir: params.outdir,
+                    seed: "tool\tpearson_corr") {
+                        row -> ["pearson.tsv", row.join("\t")]
+    }
+
+    AVERAGE_TSV(ch_pearson)
+    CORRELATION_MULTIQC(AVERAGE_TSV.out.tsv)
+
+    //Jaccard Workflow
+    ch_jaccard = BEDTOOLS_JACCARD(ch_joined, [[], []]).tsv
+
+    ch_stats = ch_jaccard.splitCsv(header: true, sep: "\t")
+        .map{ meta, values -> [meta.id, values.intersection, values.union, values.jaccard, values.n_intersections]}
+        .collectFile( newLine: true,
+                        storeDir: params.outdir,
+                        seed: "tool\tintersection\tunion\tjaccard\tn_intersections") {
+                            row -> ["jaccard.tsv", row.join("\t")]
+        }
+
+    JACCARD_MULTIQC(ch_stats)
+
+    //combine results
+    ch_versions = ch_versions.mix(SORT.out.versions)
+    ch_versions = ch_versions.mix(BEDTOOLS_MERGE.out.versions)
+    ch_versions = ch_versions.mix(BEDTOOLS_JACCARD.out.versions)
+    ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV.out.versions)
+    ch_versions = ch_versions.mix(BEDTOOLS_INTERSECT.out.versions)
+    ch_versions = ch_versions.mix(JACCARD_MULTIQC.out.versions)
+    ch_versions = ch_versions.mix(AVERAGE_TSV.out.versions)
+    ch_versions = ch_versions.mix(OVERLAP_PLOT.out.versions)
+    ch_versions = ch_versions.mix(OVERLAP_JSON.out.versions)
+    ch_versions = ch_versions.mix(LOCATION_PLOT.out.versions)
+    ch_versions = ch_versions.mix(LOCATION_JSON.out.versions)
+    ch_versions = ch_versions.mix(SEQ_DEPTH_CORRELLATION.out.versions)
+
+    ch_versions = ch_versions.mix(CORRELATION_MULTIQC.out.versions)
+
+    ch_reports = JACCARD_MULTIQC.out.report.mix(LOCATION_JSON.out.report)
+    ch_reports = ch_reports.mix(OVERLAP_JSON.out.report)
+    ch_reports = ch_reports.mix(CORRELATION_MULTIQC.out.report)
```

Comment:
Better move this `mix` operations to the locations where each tool is called instead of having one big junk